modules/pxp-module-puppet

```diff
@@ -32,13 +32,19 @@ env_fix_up = nil
 # use this way of defining the `get_env_fix_up` method so that it can "see"
 # the `env_fix_up` variable
 self.class.send(:define_method, :get_env_fix_up) do
+  # If running in a C or POSIX locale, ask Puppet to use UTF-8
+  base_env = {}
+  if Encoding.default_external == Encoding::US_ASCII
```

Comment:
@Iristyle updated. The theory with this version is that: when Ruby's default encoding is US-ASCII, Puppet may still use other character encodings from the manifest. Request that characters are printed as UTF-8 (to hopefully handle any encodings Puppet may deal with). Later, try to change all execution output to UTF-8 (because that's what we want to compare against with regexes anyway), and if that fails assume we have UTF-8 characters but used a non-UTF-8 locale.

There may be some un-handled corner-cases, but it shouldn't make them worse. If you try to run with a German codepage on Windows, but a manifest has UTF-8 characters that can't be represented in it, you already have a problem.